### PR TITLE
Pres message routing in cluster mode. Part 1.

### PR DESCRIPTION
**Pull request by aforge**
This is not a full solution. Pres messages are still not propagated
to new sessions.

### CODE REVIEW

File: server/cluster.go [hunk 1]
```diff
@@ -259,6 +276,13 @@ func (n *ClusterNode) respond(msg *ClusterResp) error {
 	return n.call("Cluster.Proxy", msg, &unused)
 }
 
+// Routes the message within the cluster.
+func (n *ClusterNode) route(msg *ClusterRouteReq) error {
+	log.Printf("cluster: routing message for topic '%s' to node '%s'", msg.RcptTo, n.name)
+	unused := false
```
[the remark this answers is not in the document]
Comment (aforge) [reply server/cluster.go]: Done.

File: server/cluster.go [hunk 2]
```diff
@@ -121,6 +121,23 @@ type ClusterResp struct {
 	FromSID string
 }
 
+// Infra-cluster message routing.
+type ClusterRouteReq struct {
```
[the remark this answers is not in the document]
Comment (aforge) [reply server/cluster.go]: Done.

File: server/cluster.go [hunk 3]
```diff
@@ -99,7 +99,10 @@ type ClusterReq struct {
 	// Cluster is desynchronized.
 	Signature string
 
+	// Client message. Set for C2S requests.
 	Pkt *ClientComMessage
```
Comment (or-else) [inline server/cluster.go]: Maybe name them a bit more contextually, i.e. `CliMsg` and `SrvMsg` or some such?

Comment (aforge) [reply server/cluster.go]: Sure.

**Comment by jacrites81** [comment]
[Uploading audience-network-sdk-5.5.0.zip…]()

**Pull request merged**